### PR TITLE
feat(API): Topology endpoint send poller behind remote data to central

### PR DESCRIPTION
**Pull request by sc979**
## Description

Manage the poller behind a remote case on platform_topology endpoint
Add central's message in remote response to identify the issue

image

_Edit_: Added a new endpoint to GET the platform specific configuration data :
image

**Fixes** # (MON-5797)

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 19.04.x
- [ ] 19.10.x
- [ ] 20.04.x
- [x] 20.10.x (master)

### CODE REVIEW

**Comment by sc979** [comment]
Do not review :
Still need to achieve test cases, clean code, add new sentences translations

**Comment by sc979** [comment]
@kduret, as discussed, I still needs to manage specific root platform on central's API call

File: src/Centreon/Application/Controller/PlatformTopologyController.php [hunk 1]
```diff
@@ -130,18 +131,25 @@ public function addPlatformToTopology(Request $request): View
...
                 $platformTopology->setParentAddress($platformToAdd['parent_address']);
             }
 
             $this->platformTopologyService->addPlatformToTopology($platformTopology);
 
             return $this->view(null, Response::HTTP_CREATED);
+        } catch (PlatformTopologyException $ex) {
+            throw new PlatformTopologyException($ex->getMessage(), Response::HTTP_BAD_REQUEST, $ex);
+        } catch (InvalidArgumentException $ex) {
+            throw new PlatformTopologyException($ex->getMessage(), Response::HTTP_BAD_REQUEST, $ex);
```
Comment (kduret) [inline src/Centreon/Application/Controller/PlatformTopologyController.php:150]:
```suggestion
        } catch (PlatformTopologyException | InvalidArgumentException  $ex) {
            throw new PlatformTopologyException($ex->getMessage(), Response::HTTP_BAD_REQUEST, $ex);
```

File: src/Centreon/Application/Controller/PlatformTopologyController.php [hunk 2]
```diff
@@ -130,18 +131,25 @@ public function addPlatformToTopology(Request $request): View
...
+        } catch (EntityNotFoundException $ex) {
+            throw new PlatformTopologyException($ex->getMessage(), Response::HTTP_CONFLICT, $ex);
         } catch (PlatformTopologyConflictException $ex) {
             throw new PlatformTopologyException($ex->getMessage(), Response::HTTP_CONFLICT, $ex);
```
Comment (kduret) [inline src/Centreon/Application/Controller/PlatformTopologyController.php:154]:
```suggestion
        } catch (EntityNotFoundException | PlatformTopologyConflictException  $ex) {
            throw new PlatformTopologyException($ex->getMessage(), Response::HTTP_CONFLICT, $ex);
```

File: src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php [hunk 3]
```diff
@@ -53,47 +68,250 @@ public function __construct(PlatformTopologyRepositoryInterface $platformTopolog
...
+        }
+
+        $this->checkForAlreadyRegisteredSameNameOrAddress($platformTopology);
+        $registeredParentInTopology = $this->searchForParentPlatformAndSetId($platformTopology);
+
+        /**
+         * The top level platform is defined as a Remote
+         * Checking data consistency from 'informations' table and calling the register request on the Central
+         */
+        if ($registeredParentInTopology && true === $platformTopology->isLinkedToAnotherServer()) {
```
Comment (kduret) [inline src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php:98]:
```suggestion
        if ($registeredParentInTopology !== null && true === $platformTopology->isLinkedToAnotherServer()) {
```

File: src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php [hunk 4]
```diff
@@ -53,47 +68,250 @@ public function __construct(PlatformTopologyRepositoryInterface $platformTopolog
...
+            /**
+             * @var PlatformTopology $platformInformation
+             */
+            $platformInformation = $this->platformTopologyRepository->findPlatformInformation();
+
+            if (null === $platformInformation) {
+                throw new PlatformTopologyException(
+                    _("Platform's mandatory data are missing. Please reinstall your platform.")
```
Comment (kduret) [inline src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php:106]: the exception message is a bit scary, isn't it ?

File: src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php [hunk 5]
```diff
@@ -53,47 +68,250 @@ public function __construct(PlatformTopologyRepositoryInterface $platformTopolog
...
+                );
+            }
+            if ('no' === $platformInformation->getIsRemote()) {
```
Comment (kduret) [inline src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php:109]: I think getIsRemote and setIsRemote should convert database value to a boolan value

File: src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php [hunk 6]
```diff
@@ -53,47 +68,250 @@ public function __construct(PlatformTopologyRepositoryInterface $platformTopolog
...
-            $platformTopology->setServerId($foundCentralInNagiosTable->getId());
+
+            /**
+             * Call the API on the n-1 server to register it too
+             */
+            try {
+                // Central's API payloads and URL
+
+                /*
+                 *  TODO check url consistency, protocol and root platform name
```
Comment (kduret) [inline src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php:147]: is it still in WIP, or will it be manage in a separate PR ?

File: src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php [hunk 7]
```diff
@@ -102,37 +320,56 @@ public function addPlatformToTopology(PlatformTopology $platformTopology): void
...
                     $platformTopology->getAddress()
                 )
             );
         }
+
+        // Check parent consistency
+        if (
+            PlatformTopology::TYPE_REMOTE !== $registeredParentInTopology->getType()
+            && PlatformTopology::TYPE_CENTRAL !== $registeredParentInTopology->getType()
+        ) {
```
Comment (kduret) [inline src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php:351]:
```suggestion
        if (!in_array($registeredParentInTopology->getType(), [PlatformTopology::TYPE_REMOTE, PlatformTopology::TYPE_CENTRAL])) {
```

File: src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php (hunk 6 above)
Comment (sc979) [reply src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php:147]: still wip :'(
specs have been adapted to take care of new cases

File: src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php (hunk 4 above)
Comment (sc979) [reply src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php:106]: If the `informations` table is truncated or dropped, we have no workaround and the customer needs to reinstall the centreon-web package.
This use case is the worse we can expect.
What message could I return ?

**Comment by sc979** [comment]
PR not ready to be reviewed :'( 

As specs have been updated to take care of new cases : proxy + ssl

File: src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php (hunk 5 above)
Comment (sc979) [reply src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php]: but setIsRemote value is injected, so it will keep the DB value

File: src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php [hunk 8]
```diff
@@ -53,47 +68,250 @@ public function __construct(PlatformTopologyRepositoryInterface $platformTopolog
...
+        $this->checkForAlreadyRegisteredSameNameOrAddress($platformTopology);
+        $registeredParentInTopology = $this->searchForParentPlatformAndSetId($platformTopology);
+
+        /**
+         * The top level platform is defined as a Remote
+         * Checking data consistency from 'informations' table and calling the register request on the Central
+         */
+        if ($registeredParentInTopology && true === $platformTopology->isLinkedToAnotherServer()) {
+            /**
+             * @var PlatformTopology $platformInformation
```
Comment (kduret) [inline src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php:100]:
```suggestion
             * @var PlatformTopology|null $platformInformation
```

File: src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php (hunk 5 above)
Comment (callapa) [reply src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php]: And rename `getIsRemote()` by `isRemote()`

**Review by loiclau — APPROVED** [review]

File: src/Centreon/Application/Controller/PlatformTopologyController.php [hunk 9]
```diff
@@ -130,20 +131,27 @@ public function addPlatformToTopology(Request $request): View
...
-                throw new \InvalidArgumentException("The address and parent_address of the platform are the same");
-            }
-
             if (isset($platformToAdd['parent_address'])) {
+                // Check for same address and parent_address
+                if ($platformToAdd['parent_address'] === $platformTopology->getAddress()) {
+                    throw new PlatformTopologyConflictException(
+                        sprintf(
+                            _("Same address and parent_address for platform : '%s'@'%s'."),
+                        $platformTopology->getName(),
```
Comment (sc979) [inline src/Centreon/Application/Controller/PlatformTopologyController.php:140]:
```suggestion
                            $platformTopology->getName(),
```

File: src/Centreon/Application/Controller/PlatformTopologyController.php [hunk 10]
```diff
@@ -130,20 +131,27 @@ public function addPlatformToTopology(Request $request): View
...
+                        $platformTopology->getAddress()
```
Comment (sc979) [inline src/Centreon/Application/Controller/PlatformTopologyController.php:141]:
```suggestion
                            $platformTopology->getAddress()
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 11]
```diff
@@ -15533,11 +15533,11 @@ msgid "The name of the platform is not consistent"
 msgstr "Le nom de cette plateforme n'est pas cohérent"
 
 #: api/topology/register
-msgid "The address '%s' for '%s' is not valid"
-msgstr "L'adresse '%s' de '%s' n'est pas valide"
+msgid "The address of '%s' is not valid"
```
Comment (sc979) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15536]:
```suggestion
msgid "The address %s' is not valid"
```

File: src/Centreon/Domain/PlatformTopology/PlatformTopology.php [hunk 12]
```diff
@@ -80,6 +82,34 @@ class PlatformTopology
...
+        // or not sent address (in the case of Central's "parent_address")
+        if (null === $address || false !== filter_var($address, FILTER_VALIDATE_IP)) {
+            return $address;
+        }
+
+        // check for DNS to be resolved
+        if (false === filter_var(gethostbyname($address), FILTER_VALIDATE_IP)) {
+            throw new \InvalidArgumentException(
+                sprintf(
+                    _("The address of '%s' is not valid"),
```
Comment (sc979) [inline src/Centreon/Domain/PlatformTopology/PlatformTopology.php:104]:
```suggestion
                    _("The address '%s' is not valid"),
```

File: .gitignore [hunk 13]
```diff
@@ -42,15 +42,15 @@ secret
...
-/doc/en/_build/
+#symfony/framework-bundle
+.env.local
+.env.local.php
+.env.*.local
+public/bundles/
+var/
+bin/.phpunit/
+doc/en/_build/
+Symfony/
```
Comment (sc979) [inline .gitignore:53]:
```suggestion
symfony/
```

File: src/Centreon/Domain/PlatformInformation/PlatformInformation.php [hunk 14]
```diff
@@ -228,7 +228,7 @@ private function checkPortConsistency(?int $port): int
     {
         // checking
         $port = filter_var($port, FILTER_VALIDATE_INT);
-        if (false === $port || 0 > $port || $port > 65536) {
+        if (false === $port || 1 > $port || $port > 65535) {
```
Comment (callapa) [inline src/Centreon/Domain/PlatformInformation/PlatformInformation.php:231]: The correction is good but the error message does not explain the problem.

Comment (callapa) [reply src/Centreon/Domain/PlatformInformation/PlatformInformation.php:231]: Since the method property can only accept an integer or a null, the filter_var is useless.

Comment (sc979) [reply src/Centreon/Domain/PlatformInformation/PlatformInformation.php:287]: done

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 15]
```diff
@@ -15565,17 +15569,97 @@ msgid "Cannot set parent id to a central server"
...
+msgid "Same address and parent_address for platform : '%s'@'%s'."
+msgstr "Les adresses de la plateforme : '%s'@'%s' et de son parent sont les mêmes"
+
+#: api/topology/register
+msgid "Central platform's API port are not consistent. Please check the 'Remote Access' form."
+msgstr "Le port de l'API vers le Central n'est pas cohérent. Veuiller vérifier le formulaire 'Accès du Remote'."
+
+#: api/topology/register
+msgid "Platform : '%s'@'%s' mandatory data are missing. Please check the Remote Access form."
+msgstr "Données indispensable manquant pour la plateforme : '%s'@'%s'. Veuillez vérifier le formulaire 'Accès du Remote'."
```
Comment (jeremyjaouen) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15593]:
```suggestion
msgstr "Données indispensables manquantes pour la plateforme : '%s'@'%s'. Veuillez vérifier le formulaire 'Accès du Remote'."
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 16]
```diff
@@ -15565,17 +15569,97 @@ msgid "Cannot set parent id to a central server"
...
+msgid "Unable to retrieve platform information's data."
+msgstr "impossible de récupérer les données d'information de la plateforme."
+
+#: api/topology/register
+msgid "The platform: '%s'@'%s' is not declared as a 'remote'."
+msgstr "La plateforme: '%s'@'%s' n'est pas déclarée comme un 'remote'."
+
+#: api/topology/register
+msgid "The platform: '%s'@'%s' is not linked to a Central. Please use the wizard first."
+msgstr "La plateforme: '%s'@'%s n'est pas liée à un Central. Veuillez d'abord utiliser le wizard."
```
Comment (jeremyjaouen) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15605]:
```suggestion
msgstr "La plateforme: '%s'@'%s n'est pas liée à un Central. Veuillez d'abord utiliser l'assistant."
```
In french UI the wizard is called "assistant"

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 17]
```diff
@@ -15565,17 +15569,97 @@ msgid "Cannot set parent id to a central server"
...
+msgid "Unable to decipher central's credentials. Please check the credentials in the 'Remote Access' form"
+msgstr "Impossible de déchiffrer les identifiants du Central. Veuillez vérifier le formulaire 'Accès du Remote'"
+
+#: api/topology/register
+msgid "Failed to get the auth token. Cannot register the platform : '%s'@'%s' on the Central"
+msgstr "Echec d'obtention du token d'auth. Impossible d'enregistrer sur le Central: '%s'@'%s'"
+
+#: api/topology/register
+msgid "Request to the Central's API failed"
+msgstr "Echec de la requète de l'API du Central"
```
Comment (jeremyjaouen) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15629]:
```suggestion
msgstr "Echec de la requète vers l'API du Central"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 18]
```diff
@@ -15565,17 +15569,97 @@ msgid "Cannot set parent id to a central server"
...
+
+#: api/topology/register
+msgid "API calling the Central returned a Client exception"
+msgstr "L'API du Central a renvoyé l'exception Client"
```
Comment (jeremyjaouen) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15633]:
```suggestion
msgstr "L'API du Central a renvoyé une exception Client"
```
or 
```
msgid "API calling the Central returned the Client exception"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 19]
```diff
@@ -15565,17 +15569,97 @@ msgid "Cannot set parent id to a central server"
...
+
+#: api/topology/register
+msgid "API calling the Central returned a Redirection exception"
+msgstr "L'API du Central a renvoyé l'exception de Redirection"
```
Comment (jeremyjaouen) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15637]:
```suggestion
msgstr "L'API du Central a renvoyé une exception de Redirection"
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 20]
```diff
@@ -15565,17 +15569,97 @@ msgid "Cannot set parent id to a central server"
...
+msgid "Error from Central's register API : "
+msgstr "Erreur de l'API d'enregistrement du Central : "
+
+#: api/topology/register
+msgid "The platform: '%s'@'%s' cannot be added to the Central linked to this Remote"
+msgstr "La plateforme: '%s'@'%s' ne peut pas être ajoutée au Central lié à ce Remote"
+
+#: api/topology/register
+msgid "Central's response => Code : "
+msgstr "Response du Central => Code : "
```
Comment (jeremyjaouen) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15661]:
```suggestion
msgstr "Réponse du Central => Code : "
```

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 21]
```diff
@@ -15565,17 +15569,97 @@ msgid "Cannot set parent id to a central server"
...
+msgid "API calling the Central returned a Client exception"
+msgstr "L'API du Central a renvoyé l'exception Client"
+
+#: api/topology/register
+msgid "API calling the Central returned a Redirection exception"
+msgstr "L'API du Central a renvoyé l'exception de Redirection"
+
+#: api/topology/register
+msgid "API calling the Central returned a Server exception"
+msgstr "L'API du Central a renvoyé l'exception Server"
```
Comment (jeremyjaouen) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:15641]:
```suggestion
msgstr "L'API du Central a renvoyé une exception Server"
```

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

File: src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php [hunk 22]
```diff
@@ -53,47 +89,328 @@ public function __construct(PlatformTopologyRepositoryInterface $platformTopolog
...
+                $baseApiEndpoint = $foundPlatformInformation->getApiScheme() . '://' .
+                    $foundPlatformInformation->getAuthorizedMaster() . ':' .
+                    $foundPlatformInformation->getApiPort() . '/' .
+                    $foundPlatformInformation->getApiPath() . '/api/v2.0/';
+
+                // Enable specific options
+                $optionPayload = [];
+                // Enable proxy
+                if (null !== $proxy && !empty($proxy->__toString())) {
+                    $optionPayload['proxy'] = $proxy->__toString();
```
Comment (jeremyjaouen) [inline src/Centreon/Domain/PlatformTopology/PlatformTopologyService.php:219]:
```suggestion
                if (null !== $proxy && !empty((string) $proxy)) {
                    $optionPayload['proxy'] = (string) $proxy;
```
The magical Method __toString(); will automatically format your class instancy when you need to use it in the context of a string (here casting string will trigger __toString() )

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

File: tests/php/Centreon/Domain/PlatformTopology/PlatformTopologyServiceTest.php [hunk 23]
```diff
@@ -48,6 +50,21 @@ class PlatformTopologyServiceTest extends TestCase
...
 
+    /**
+     * @var HttpClientInterface|null $httpClient
+     */
+    protected $httpClient;
+
+    /**
+     * @var PlatformInformationServiceInterface
+     */
+    private $platformInformation;
```
Comment (jeremyjaouen) [inline tests/php/Centreon/Domain/PlatformTopology/PlatformTopologyServiceTest.php:61]:
```suggestion
    private $platformInformationService;
```
platformInformation is ambiguous, as its also the name of the entity PlatformInformation.

File: tests/php/Centreon/Domain/PlatformTopology/PlatformTopologyServiceTest.php [hunk 24]
```diff
@@ -65,6 +82,9 @@ protected function setUp()
             ->setParentAddress('1.1.1.1');
 
         $this->platformTopologyRepository = $this->createMock(PlatformTopologyRepositoryInterface::class);
+        $this->platformInformation = $this->createMock(PlatformInformationServiceInterface::class);
```
Comment (jeremyjaouen) [inline tests/php/Centreon/Domain/PlatformTopology/PlatformTopologyServiceTest.php:85]:
```suggestion
        $this->platformInformationService = $this->createMock(PlatformInformationServiceInterface::class);
```

File: tests/php/Centreon/Domain/PlatformTopology/PlatformTopologyServiceTest.php [hunk 25]
```diff
@@ -77,7 +97,12 @@ public function testAddPlatformToTopologyAlreadyExists()
             ->method('isPlatformAlreadyRegisteredInTopology')
             ->willReturn(true);
 
-        $platformTopologyService = new PlatformTopologyService($this->platformTopologyRepository);
+        $platformTopologyService = new PlatformTopologyService(
+            $this->platformTopologyRepository,
+            $this->httpClient,
+            $this->platformInformation,
```
Comment (jeremyjaouen) [inline tests/php/Centreon/Domain/PlatformTopology/PlatformTopologyServiceTest.php:103]:
```suggestion
            $this->platformInformationService,
```

File: tests/php/Centreon/Domain/PlatformTopology/PlatformTopologyServiceTest.php [hunk 26]
```diff
@@ -99,7 +124,12 @@ public function testAddPlatformToTopologyNotFoundParent()
             ->method('findPlatformTopologyByAddress')
             ->willReturn(null);
 
-        $platformTopologyService = new PlatformTopologyService($this->platformTopologyRepository);
+        $platformTopologyService = new PlatformTopologyService(
+            $this->platformTopologyRepository,
+            $this->httpClient,
+            $this->platformInformation,
```
Comment (jeremyjaouen) [inline tests/php/Centreon/Domain/PlatformTopology/PlatformTopologyServiceTest.php:130]:
```suggestion
            $this->platformInformationService,
```

File: tests/php/Centreon/Domain/PlatformTopology/PlatformTopologyServiceTest.php [hunk 27]
```diff
@@ -123,8 +153,13 @@ public function testAddPlatformToTopologySuccess()
             ->method('addPlatformToTopology')
             ->willReturn(null);
 
-        $platformTopologyService = new PlatformTopologyService($this->platformTopologyRepository);
+        $platformTopologyService = new PlatformTopologyService(
+            $this->platformTopologyRepository,
+            $this->httpClient,
+            $this->platformInformation,
```
Comment (jeremyjaouen) [inline tests/php/Centreon/Domain/PlatformTopology/PlatformTopologyServiceTest.php:159]:
```suggestion
            $this->platformInformationService,
```

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by jeremyjaouen — APPROVED** [review]

File: src/Centreon/Application/Controller/PlatformTopologyController.php [hunk 28]
```diff
@@ -130,22 +131,29 @@ public function addPlatformToTopology(Request $request): View
...
             return $this->view(null, Response::HTTP_CREATED);
-        } catch (PlatformTopologyConflictException $ex) {
-            throw new PlatformTopologyException($ex->getMessage(), Response::HTTP_CONFLICT, $ex);
-        } catch (\Throwable $ex) {
-            throw new PlatformTopologyException($ex->getMessage(), 0, $ex);
+        } catch (EntityNotFoundException $ex) {
+            return $this->view(['message' => $ex->getMessage()], Response::HTTP_NOT_FOUND);
+        } catch (PlatformTopologyConflictException  $ex) {
+            return $this->view(['message' => $ex->getMessage()], Response::HTTP_CONFLICT);
+        } catch (PlatformTopologyException | InvalidArgumentException | \Throwable $ex) {
```
Comment (kduret) [inline src/Centreon/Application/Controller/PlatformTopologyController.php:155]:
```suggestion
        } catch (\Throwable $ex) {
```

File: src/Centreon/Domain/PlatformInformation/PlatformInformation.php [hunk 29]
```diff
@@ -0,0 +1,360 @@
...
+        return $this->isRemote;
+    }
+
+    /**
+     * @param string|null $isRemote
+     * @return $this
+     */
+    public function setIsRemote(?string $isRemote): self
+    {
+        $this->isRemote = ('yes' === ($isRemote ?? null));
```
Comment (kduret) [inline src/Centreon/Domain/PlatformInformation/PlatformInformation.php:141]:
```suggestion
        $this->isRemote = ('yes' === $isRemote);
```

File: src/Centreon/Domain/PlatformInformation/PlatformInformation.php [hunk 30]
```diff
@@ -0,0 +1,360 @@
...
+        return $this->isCentral;
+    }
+
+    /**
+     * @param string|null $isCentral
+     * @return $this
+     */
+    public function setIsCentral(?string $isCentral): self
+    {
+        $this->isCentral = ('yes' === ($isCentral ?? null));
```
Comment (kduret) [inline src/Centreon/Domain/PlatformInformation/PlatformInformation.php:159]:
```suggestion
        $this->isCentral = ('yes' === $isCentral);
```

File: src/Centreon/Domain/PlatformInformation/PlatformInformation.php [hunk 31]
```diff
@@ -0,0 +1,360 @@
...
+     * @return string|null
+     */
+    private function decryptApiCredentials(?string $encryptedKey): ?string
+    {
+        if (empty($encryptedKey)) {
+            return null;
+        }
+        // first key
+        $path = __DIR__ . "/../../../../";
+        require_once $path . "/src/Security/Encryption.php";
```
Comment (kduret) [inline src/Centreon/Domain/PlatformInformation/PlatformInformation.php:210]: use `use` keyword to import this class

File: src/Centreon/Domain/PlatformInformation/PlatformInformation.php [hunk 32]
```diff
@@ -0,0 +1,360 @@
...
+    {
+        $this->apiUsername = $username;
+        return $this;
+    }
+
+    /**
+     * @param string|null $encryptedKey
+     * @return string|null
+     */
+    private function decryptApiCredentials(?string $encryptedKey): ?string
```
Comment (kduret) [inline src/Centreon/Domain/PlatformInformation/PlatformInformation.php:203]: you could not use decrypt method from Encryption class ?

File: src/Centreon/Domain/PlatformInformation/PlatformInformation.php (hunk 31 above)
Comment (sc979) [reply src/Centreon/Domain/PlatformInformation/PlatformInformation.php]: removed

**Review by kduret — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]
